app/scripts/directives/fromFile.js

```diff
@@ -178,24 +178,25 @@ angular.module("openshiftConsole")
             };
             return false;
           }
-          if (!item.metadata) {
-            $scope.error = {
-              message: "Resource is missing metadata field."
-            };
-            return false;
-          }
-          // Validate if resource has 'name' field in its 'metadata', but it's not a List
-          if (!item.metadata.name && !item.kind.endsWith("List")) {
-            $scope.error = {
-              message: "Resource name is missing in metadata field."
-            };
-            return false;
-          }
-          if (item.metadata.namespace && item.metadata.namespace !== $scope.projectName) {
-            $scope.error = {
-              message: item.kind + " " + item.metadata.name + " can't be created in project " + item.metadata.namespace + ". Can't create resource in different projects."
-            };
-            return false;
+          if (!item.kind.endsWith("List")) {
```

Comment:
I'd rather return early here.

```js
if (item.kind.endsWith("List")) {
  // Nothing else to check for lists.
  return true;
}
```

Instead of having a large `if` block. It's easier to reason about this way.